docs/proposals/kuratorcd/kurator-cd.md

```diff
@@ -0,0 +1,622 @@
+
+```console
+// Note: refer to https://github.com/fluxcd/flagger/blob/main/pkg/apis/flagger/v1beta1/canary.go
+type DeliveryPolicy struct {
+    // Kurator only supports istio for now.
+    // New Provider will be added later.
+    // +optional
+    TrafficRoutingProvider string `json:"trafficRoutingProvider,omitempty"`
+
+    // TargetObject specifies what object to deploy the test to. 
+    // Objects of type deployment or daemonSet.
+    TargetObject *TargetObjectReference `json:"targetObject"`
+
+    // ServiceCfg is the configuration of the service pointing to the TargetObject.
+    ServiceCfg *ServiceConfig `json:"serviceCfg"`
+
+    // The virtualserviceCfg defines the configuration of the gateway, traffic distribution rules, and so on.
+    VirtualServiceCfg *VirtualServiceConfig `json:"virtualServiceCfg"`
+
+    // TrafficAnalysis defines the validation process of a release
+    TrafficAnalysis *TrafficAnalysis `json:"trafficAnalysis,omitempty"`
+
+    // ProgressDeadlineSeconds represents the maximum time in seconds for a
+    // canary deployment to make progress before it is considered to be failed.
+    // Defaults to 600s.
+    // +optional
+    ProgressDeadlineSeconds *int32 `json:"progressDeadlineSeconds,omitempty"`
+
+    // SkipAnalysis promotes the canary without analysing it
+    // +optional
+    SkipTrafficAnalysis bool `json:"skipTrafficAnalysis,omitempty"`
+
+    // Restore resources to initial state when deleting canary resources.
+    // Use of the revertOnDeletion property should be enabled
+    // when you no longer plan to rely on Kurator for deployment management.
+    // +optional
+    RevertOnDeletion bool `json:"revertOnDeletion,omitempty"`
+
+    // Suspend, if set to true will suspend the Canary, disabling any canary runs
+    // regardless of any changes to its target, services, etc. Note that if the
+    // Canary is suspended during an analysis, its paused until the Canary is unsuspended.
+    // +optional
+    Suspend bool `json:"suspend,omitempty"`
+}
+```
+
+TargetObjectReference contains enough information to let you locate the typed referenced object in the same namespace. The two types of Kind now supported are `Deployment` and `DaemonSet`.
+
+```concole
+// Note: refer to https://github.com/fluxcd/flagger/blob/main/pkg/apis/flagger/v1beta1/canary.go
+type TargetObjectReference struct {
+    // API version of the referent
+    // +optional
+    APIVersion string `json:"apiVersion,omitempty"`
+
+    // Kind of the referent
+    // +optional
+    Kind string `json:"kind,omitempty"`
+
+    // Name of the referent
+    Name string `json:"name"`
+}
+```
+
+The ServiceConfig is the configuration of the service that is generated according to the object specified in the TargetObject. The main thing is to specify the `Port` and `targetPort`.
+
+```console
+// Note: refer to https://github.com/fluxcd/flagger/blob/main/pkg/apis/flagger/v1beta1/canary.go
+type ServiceConfig struct {
```

Comment:
is this used for canary svc generation? If so why does not it use the original svc's spec.